spec/models/calculation_result_spec.rb

```diff
@@ -4,4 +4,90 @@
   it "raises an error if there's an unknown result" do
     expect { described_class.new({}).decision }.to raise_error("Unhandled CFE result: ")
   end
+
+  describe "#client_main_home_rows" do
+    it "does not show cost of sale deduction row in situations where cost of sale deduction is zero" do
+      data = {
+        assessment: {
+          capital: {
+            capital_items: {
+              properties: {
+                main_home: {
+                  value: 0,
+                  outstanding_mortgage: 0,
+                  main_home_equity_disregard: 0,
+                  transaction_allowance: 0,
+                },
+              },
+            },
+          },
+        },
+      }
+
+      expect(described_class.new(data).client_main_home_rows.keys).not_to include(:deductions)
+    end
+
+    it "does shows cost of sale deduction row in situations where cost of sale deduction is not zero" do
+      data = {
+        assessment: {
+          capital: {
+            capital_items: {
+              properties: {
+                main_home: {
+                  value: 0,
+                  outstanding_mortgage: 0,
+                  main_home_equity_disregard: 0,
+                  transaction_allowance: 1,
+                },
+              },
+            },
+          },
+        },
+      }
+
+      expect(described_class.new(data).client_main_home_rows.keys).to include(:deductions)
+    end
+
+    describe "#client_additional_property_rows" do
+      it "does not show cost of sale deduction row in situations where cost of sale deduction is zero" do
+        data = {
+          assessment: {
+            capital: {
+              capital_items: {
+                properties: {
+                  additional_properties: [{
+                    value: 0,
+                    outstanding_mortgage: 0,
+                    transaction_allowance: 0,
+                  }],
+                },
+              },
+            },
+          },
+        }
```

Comment:
IMHO a FactoryBot helper would remove this noise

Reply:
Good idea - I've rebased this PR against EL-712 to give me use of the FactoryBot stuff I added there (although that rebase has slightly screwed up the commit history)